codecs/h264_packet_test.go

```diff
@@ -291,3 +292,35 @@ func TestH264Payloader_Payload_SPS_and_PPS_handling(t *testing.T) {
 		t.Fatal("SPS and PPS aren't packed together")
 	}
 }
+
+func TestH264Payloader_PayloadAndDetectDataIframeAfterSpsPps(t *testing.T) {
+	pck := H264Payloader{}
+
+	// When packetizing SPS and PPS are emitted with following NALU
+	res := pck.Payload(1500, []byte{0x07, 0x00, 0x01})
+	if len(res) != 0 {
+		t.Fatal("Generated payload should be empty")
+	}
+
+	res = pck.Payload(1500, []byte{0x08, 0x02, 0x03})
+	if len(res) != 0 {
+		t.Fatal("Generated payload should be empty")
+	}
+
+	idx := -1
+	res, idx = pck.PayloadAndDetectData(1500, []byte{0x05, 0x04, 0x05})
+
+	if idx != 1 {
```

Comment:
Took me some time to understand why this is 1 and not 2.
Can you add better comment which says that both SPS and PPS are emitted in a single payload?
So we end up with
res[0] -> SPS+PPS
res[1] -> IFRAME

Reply:
that's pretty clear when you read the payloader and there is a dedicated test for that just a bit earlier:
TestH264Payloader_Payload_SPS_and_PPS_handling

Reply:
that's pretty clear when you read the payloader and there is a dedicated test for that just a bit earlier:
TestH264Payloader_Payload_SPS_and_PPS_handling